tests/drivers/display/display_read_write/testcase.yaml

```diff
@@ -50,3 +50,15 @@ tests:
     extra_configs:
       - CONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_BGR_565=y
       - CONFIG_SDL_DISPLAY_USE_HARDWARE_ACCELERATOR=n
+  drivers.display.read_write.ili9340:
+    tags:
+      - shield
+    extra_args:
+      - SHIELD=adafruit_2_8_tft_touch_v2
+      - EXTRA_DTC_OVERLAY_FILE=adafruit_2_8_tft_touch_rgb565.overlay
+    extra_configs:
+      - CONFIG_ILI9XXX_READ=y
+    # Use platform_allow, because we cannot filter the test based on the
+    # presence of the arduino_spi DT nodelabel
```

Comment:
why is this comment necessary

Reply:
What harm does it cause? It's an explanation as to why we are using platform_allow (which IMO, is rarely the best way to filter a given test). Generally, I try to avoid using platform_allow for generic tests unless there is a good reason- this comment is intended to provide that justification

Reply:
The other test cases don't appear to me to filter for this nodelabel

Reply:
Yes, because those test cases aren't using a shield. Since we need to add a shield to define the ili9xxx display controller, we need to be sure that there is an 'arduino_spi' node present for the shield overlay to reference. Since twister can only filter after generating the devicetree based on the cmake arguments (as far as I know), we can't filter based on the presence of an 'arduino_spi' nodelabel (because the devicetree phase would simply fail if there wasn't one)- hence why we are using platform_allow